Messaging.md

```diff
@@ -339,80 +371,182 @@ Witnesses should be sorted by the `input_serial_id` sent in `funding_input` defi
 Two types of events are described, see [the oracle specification](./Oracle.md#event-descriptor) for more details.
 
-#### Version 0 `enum_event_descriptor`
+#### `enum_event_descriptor`
 
-1. type: 55302 (`enum_event_descriptor_v0`)
-2. data:
-   * [`u16`:`num_outcomes`]
-   * [`string`:`outcome_1`]
-   * ...
-   * [`string`:`outcome_n`]
+1. implements: `event_descriptor`
+1. type: 0
+1. data:
+   * [`bigsize`:`num_outcomes`]
+   * [`num_outcomes*string`:`outcomes`]
 
 This type of event descriptor is a simple enumeration where the value `n` is the number of outcomes in the event.
 
-Note that `outcome_i` is the outcome value itself and not its hash that will be signed by the oracle.
+Note that `outcomes[i]` is the outcome value itself and not its hash that will be signed by the oracle.
 
-#### Version 0 `digit_decomposition_event_descriptor`
+#### `digit_decomposition_event_descriptor`
 
-1. type: 55306 (`digit_decomposition_event_descriptor_v0`)
-2. data:
+1. implements: `event_descriptor`
+1. type: 1
+1. data:
    * [`bigsize`:`base`]
    * [`bool`:`is_signed`]
    * [`string`:`unit`]
    * [`int32`:`precision`]
    * [`u16`:`nb_digits`]
 
+### The `oracle_event_timestamp` Type
+
+This type contains timestamp information about when an `oracle_event` is to happen.
+
+There are two kinds of `oracle_event_timestamps`: fixed timestamps for events with a
+known (expected) epoch time at which the event will occur, and range for events whose
+time is not known ahead of time and also for events which may occur at any time in the
+specified range.
+
+When using an `oracle_event_timestamp_range` for events which may occur at any time,
+the `earliest_expected_time_epoch` should correspond to the time at which the oracle
+announcement was created and `latest_expected_time_epoch` should correspond to the
+latest time for which the oracle will be responsible for attesting to the event's occurence.
+Note that it is possible to have functionally open-ended events through the use of very large
+`latest_expected_time_epoch` timestamps.
+
+#### `fixed_oracle_event_timestamp`
+
+1. implements: `oracle_event_timestamp`
+1. type: 0
+1. data:
+   * [`u32`:`expected_time_epoch`]
+
+#### `range_oracle_event_timestamp`
+
+1. implements: `oracle_event_timestamp`
+1. type: 1
+1. data:
+   * [`u32`:`earliest_expected_time_epoch`]
+   * [`u32`:`latest_expected_time_epoch`]
+
 ### The `oracle_event` Type
 
 This type contains information provided by an oracle on an event that it will attest to.
 See [the Oracle specifications](./Oracle.md#oracle-event) for more details.
 
-#### Version 0 `oracle_event`
+#### `oracle_event`
 
-1. type: 55330 (`oracle_event_v0`)
-2. data:
-   * [`u16`:`nb_nonces`]
-   * [`nb_nonces*x_point`:`oracle_nonces`]
-   * [`u32`:`event_maturity_epoch`]
+1. data:
+   * [`oracle_event_timestamp`:`timestamp`]
    * [`event_descriptor`:`event_descriptor`]
    * [`string`:`event_id`]
 
+### The `oracle_keys` Type
```

Comment:
Would it make sense to rename this to `oracle_metadata` or something else? Since it contains quite more than just keys.

Reply:
agree, makes sense to call it that